src/transforms/reduce/merge_strategy.rs

```diff
@@ -132,6 +164,118 @@ impl ReduceValueMerger for ArrayMerger {
 
 //------------------------------------------------------------------------------
 
+#[derive(Debug, Clone)]
+struct LongestArrayMerger {
+    v: Vec<Value>,
+}
+
+impl LongestArrayMerger {
+    fn new(v: Vec<Value>) -> Self {
+        Self { v }
+    }
+}
+
+impl ReduceValueMerger for LongestArrayMerger {
+    fn add(&mut self, v: Value) -> Result<(), String> {
+        if let Value::Array(a) = v {
+            if a.len() > self.v.len() {
+                self.v = a;
+            }
+        } else {
+            Err(format!(
+                "expected array value, found: '{}'",
+                v.to_string_lossy()
+            ))
+        }
+        Ok(())
+    }
+
+    fn insert_into(self: Box<Self>, k: String, v: &mut LogEvent) -> Result<(), String> {
+        v.insert(k, Value::Array(self.v));
+        Ok(())
+    }
+}
+
+//------------------------------------------------------------------------------
+
+#[derive(Debug, Clone)]
+struct ShortestArrayMerger {
+    v: Vec<Value>,
+}
+
+impl ShortestArrayMerger {
+    fn new(v: Vec<Value>) -> Self {
+        Self { v }
+    }
+}
+
+impl ReduceValueMerger for ShortestArrayMerger {
+    fn add(&mut self, v: Value) -> Result<(), String> {
+        if let Value::Array(a) = v {
+            if a.len() < self.v.len() {
+                self.v = a;
+            }
+        } else {
+            Err(format!(
+                "expected array value, found: '{}'",
+                v.to_string_lossy()
+            ))
+        }
+        Ok(())
+    }
+
+    fn insert_into(self: Box<Self>, k: String, v: &mut LogEvent) -> Result<(), String> {
+        v.insert(k, Value::Array(self.v));
+        Ok(())
+    }
+}
+
+//------------------------------------------------------------------------------
+#[derive(Debug, Clone)]
+struct UniqueMerger {
+    v: HashSet<Value>,
+}
+
+fn insert_value(h: &mut HashSet<Value>, v: Value) {
+    match v {
+        Value::Map(m) => {
+            for (_, v) in m {
+                h.insert(v);
+            }
+        }
+        Value::Array(vec) => {
+            for v in vec {
+                h.insert(v);
+            }
+        }
+        _ => {
+            h.insert(v);
+        }
+    }
+}
```

Comment:
Looking at this again, this is actually a bit different than I would have expected for `unique`. This is more like "flatten" and then "unique". What would you think of calling this option `flat_unique`? That would leave room for another `unique` that wouldn't flatten.

Reply:
Sounds good to me.